includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1792,6 +1792,7 @@ public function get_level3_data_from_order( WC_Order $order ): array {
 
 			$description     = substr( $item->get_name(), 0, 26 );
 			$quantity        = $item->get_quantity();
+			$quantity        = is_int( $quantity ) ? $quantity : 1; // In case of float type value of quantity, the value should be 1.
 			$unit_cost       = WC_Payments_Utils::prepare_amount( $subtotal / $quantity, $currency );
```

Comment:
Unit cost is calculated by diving the `$subtotal` to `$quantity`. In the case of replacing the `$quantity` with `1`, the total amounts are still correct.